### PR TITLE
Add better autoconf support for libboost

### LINKED ISSUE #302: build: configure fails to find boost libraries on Raspbian GNU/Linux 8 (jessie)
When building on this system, configure fails with
```
checking for boostlib >= 1.53.0 (105300)... yes
checking whether the Boost::System library is available... yes
configure: error: Could not find a version of the library!
```
even though the boost libraries are installed.  A workaround is to configure with
```
configure --with-boost-libdir=/usr/lib/arm-linux-gnueabihf
```

It would appear that `ax_boost_base.m4` is setting `BOOST_LDFLAGS=-L/usr/lib`, but the boost libraries (and libraries in general on this system) are in the gnueabihf directory noted above.

### PR DESCRIPTION
**Pull request by dongahn**
Add `powerpc64le` and `armv7l` platforms to `ax_boost_base.m4`
such that configure will search for the system library path where
boost RPMs install their boost libraries.

Add the actual boost library names in the config error messages
(`ax_boost_filesystem.m4`, `ax_boost_system.m4`
and `ax_boost_graph.m4`).

Add a new m4 script that checks whether the buggy libboost versions
as tested and documented at #297. 

Resolve #302,  #297 and #396.

### CODE REVIEW

**Comment by SteVwonder** [comment]
Ouch.  When I updated the dependencies in the (now old) dep builder, I probably broke something.  Sorry!  I agree that merging #392 should fix this.

**Comment by garlick** [comment]
There's a missing comma in the armv7l patch, but after I add that, configure seems to find all the boost bits!
```
checking for boostlib >= 1.53.0 (105300)... yes
checking for buggy boostlib versions (105400-105800)... no
checking whether the Boost::System library is available... yes
checking for exit in -lboost_system... yes
checking whether the Boost::Filesystem library is available... yes
checking for exit in -lboost_filesystem... yes
checking whether the Boost::Graph library is available... yes
checking for exit in -lboost_graph... yes
checking whether the Boost::Regex library is available... yes
checking for exit in -lboost_regex... yes
```

```patch
diff --git a/m4/ax_boost_base.m4 b/m4/ax_boost_base.m4
index f750fa4..c9749dc 100644
--- a/m4/ax_boost_base.m4
+++ b/m4/ax_boost_base.m4
@@ -114,7 +114,7 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
     AS_CASE([${host_cpu}],
       [x86_64],[libsubdirs="lib64 libx32 lib lib64"],
       [ppc64|s390x|sparc64|aarch64|ppc64le|powerpc64le],[libsubdirs="lib64 lib lib64"],
-      [armv7l],[libsubdirs="lib/arm-linux-gnueabihf lib"]
+      [armv7l],[libsubdirs="lib/arm-linux-gnueabihf lib"],
       [libsubdirs="lib"]
     )

```

**Comment by dongahn** [comment]
Great! I didn't have a test machine and that crept in. Sorry about that.

**Comment by garlick** [comment]
No worries!

File: configure.ac [hunk 1]
```diff
@@ -62,7 +62,9 @@ AC_DEFINE_UNQUOTED([LIBLUA_SO], ["$LIBLUA_SONAME"],
 
 PKG_CHECK_MODULES([UUID], [uuid], [], [])
 AX_FLUX_CORE
+
 AX_BOOST_BASE([1.53.0], [], [AC_MSG_ERROR([Please install boost >= 1.53)])])
```
Comment (SteVwonder) [inline configure.ac]: Can you update this error message to match the one in buggy boost? `Please use Boost == 1.53 or > 1.58`

**Comment by SteVwonder** [comment]
Other than the one nit, LGTM.  @dongahn, maybe you can take a look at #396 and see if the boost portion of that makes sense to include in this PR.  If not, I'm happy to merge this PR without it.

**Comment by dongahn** [comment]
@SteVwonder and @garlick: I addressed both of the issues you found. Also, as part of that, I addressed the issues captured in #396 as well. Thanks. 

@SteVwonder: everything turned green and from my perspective this can go in.

**Comment by SteVwonder** [comment]
Thanks @dongahn! Merging.

**Pull request merged**